external/fv3fit/fv3fit/pytorch/cyclegan/train.py

```diff
@@ -0,0 +1,145 @@
+from fv3fit._shared.hyperparameters import Hyperparameters
+import dataclasses
+import tensorflow as tf
+from fv3fit.pytorch.predict import PytorchPredictor
+from fv3fit.pytorch.loss import LossConfig
+from fv3fit.pytorch.optimizer import OptimizerConfig
+from fv3fit.pytorch.training_loop import TrainingConfig
+
+from fv3fit._shared import register_training_function
+from typing import (
+    List,
+    Optional,
+    Tuple,
+)
+from fv3fit.tfdataset import ensure_nd, apply_to_mapping
+from .network import Generator
+from fv3fit.pytorch.graph.train import (
```

Comment:
I moved the first two, I'd like to leave `get_Xy_dataset` for now because I've made significant changes to it in the follow-on PR and want to avoid merge conflicts. I will clean this up in that PR.